app/models/notification.rb

```diff
@@ -24,6 +25,11 @@ class Notification < ApplicationRecord
   scope :unread,   ->(unread)       { where(unread: unread) }
   scope :owner,    ->(owner_name)   { where(repository_owner_name: owner_name) }
 
+  scope :state,    ->(state) { joins(:subject).where('subjects.state = ?', state) }
+
+  scope :subjectable, -> { where(subject_type: ['Issue', 'PullRequest', 'Commit', 'Release']) }
```

Comment:
Currently there's one type of subject that doesn't have anything to download `RepositoryInvitation`, which has a subject url like `https://github.com/octobox/octobox/invitations`.

I can imagine GitHub adding more types in the future which is why it's listing the 4 types we know are supported by the the `#update_subject` method.